### ISSUE TITLE
[QUESTION] Keycloak Integration - Which device needs to be able to access the keycloak server?

### ISSUE THREAD

**Issue by somerandomguy17** [Lissy93/dashy#883]
### Question

TL;DR: When using the Keycloak integration for authentication, which device actually needs to be able to resolve the keycloak servers URL and communicate with the keycloak server? Is it sufficient for the server that Dashy is installed on can resolve the keycloak URL and communicate with the keycloak server, or does the initial client (i.e. the machine that dashy is being accessed from _also_ need to be able to resolve the keycloak server URL and access the server directly?

In my setup, I have Dashy installed on a server in my network's dmz _(dashy.dmz.com)_ with keycloak running on a server in a private subnet _(keycloak.internal.lan)_. The DMZ server can resolve and access _keycloak.internal.lan_ fine and the keycloak integration works perfectly if I access it from within my network (i.e. I can sign in initially, hiding/showing specific sections for users works, etc). I'm now trying to set it up for external access so I've set up a wireguard tunnel from my VPS to the dmz server, and a reverse proxy to direct traffic from the VPS to _dashy.dmz.com_. However when I do that, it appears Dashy is failing before I can get the keycloak authentication page with the "It appears something went wrong..." error.

I don't think this is a CORS error, because the exact same URL from within my network works properly (i.e. dashy can be accessed at _www.publicsite.com_ - if I'm within my network, my DNS catches it and everything works properly. If I am outside my network and go to _www.publicsite.com_, Cloudflare (my DNS provider) successfully points to my VPS, which proxies down the wireguard tunnel to _dashy.dmz.com_ but I see the errors mentioned above). I've also confirmed that disabling the keycloak authentication "resolves" the issue (i.e. I can access Dashy properly, but now I don't have authentication which I don't wnat to do for an external facing site).

Is my diagnosis correct that the issue is that the endpoint client (i.e. where I'm accessing dashy from, or perhaps even my VPS) needs to be able to resolve _keycloak.internal.lan_ and that's what's causing the issue? Or is there perhaps something else going on here

Thanks!

### Category

Authentication

### Please tick the boxes

- [X] You are using a [supported](https://github.com/Lissy93/dashy/blob/master/.github/SECURITY.md#supported-versions) version of Dashy (check the first two digits of the version number)
- [X] You've checked that this [question hasn't already been raised](https://github.com/Lissy93/dashy/issues?q=is%3Aissue)
- [X] You've checked the [docs](https://github.com/Lissy93/dashy/tree/master/docs#readme) and [troubleshooting](https://github.com/Lissy93/dashy/blob/master/docs/troubleshooting.md#troubleshooting) guide
- [X] You agree to the [code of conduct](https://github.com/Lissy93/dashy/blob/master/.github/CODE_OF_CONDUCT.md#contributor-covenant-code-of-conduct)
